plugins/modules/host_template.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2024 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ansible_collections.cloudera.cluster.plugins.module_utils.cm_utils import (
+    ClouderaManagerModule,
+)
+from cm_client import (
+    HostTemplatesResourceApi,
+    ClustersResourceApi,
+    ApiHostTemplate,
+    ApiRoleConfigGroupRef,
+    ApiClusterRef,
+    ApiHostTemplateList,
+)
+from cm_client.rest import ApiException
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: host_template
+short_description: Configure a host template
+description:
+  - Creates a new host template or updates an existing one
+  - The module supports C(check_mode).
+author:
+  - "Ronald Suplina (@rsuplina)"
+requirements:
+  - cm_client
+options:
+  name:
+    description:
+      - The associated cluster name.
+    type: str
+    required: yes
+    aliases:
+      - cluster_name
+  host_template_name:
+    description:
+      - The name of the host template.
+    type: str
+    required: yes
+  roleConfigGroupRefs:
+    description:
+      - The names of the role config groups
+    type: list
+    returned: yes
```

Comment:
Invalid doc parameter

Reply:
`returned` is not a valid parameter in the `DOCS` section -- I believe you mean `required`

Reply:
Also, `role_groups` is only required when `state=present`, so construct this logic in the `argument_spec` of the module and remove the `required` flag here. (Need to add a note in the `description`, however!)